packages/editor/src/components/collab-sidebar/utils.js

```diff
@@ -98,27 +98,38 @@ export function getCommentExcerpt( text, excerptLength = 10 ) {
  * @param {string}       additionalSelector The additional selector to focus on.
  */
 export function focusCommentThread( commentId, container, additionalSelector ) {
-	const getFocusElement = () => {
-		const commentThread = container?.querySelector(
-			`[role=listitem][id="comment-thread-${ commentId }"]`
-		);
-		if ( additionalSelector ) {
-			return commentThread?.querySelector( additionalSelector );
+	if ( ! commentId || ! container ) {
+		return;
+	}
+
+	const threadSelector = `[role=listitem][id="comment-thread-${ commentId }"]`;
+	const selector = additionalSelector
+		? `${ threadSelector } ${ additionalSelector }`
+		: threadSelector;
+
+	return new Promise( ( resolve ) => {
+		if ( container.querySelector( selector ) ) {
+			return resolve( container.querySelector( selector ) );
 		}
-		return commentThread;
-	};
 
-	let focusElement = getFocusElement();
-	if ( focusElement ) {
-		focusElement.focus();
-	} else {
-		// The element hasn't been painted yet. Defer focusing on the next frame.
-		window.requestAnimationFrame( () => {
-			focusElement = getFocusElement();
-			// Ignore if the element still doesn't exist.
-			if ( focusElement ) {
-				focusElement.focus();
+		let timer = null;
+		// Wait for the element to be added to the DOM.
+		const observer = new window.MutationObserver( () => {
+			if ( container.querySelector( selector ) ) {
+				clearTimeout( timer );
+				observer.disconnect();
+				resolve( container.querySelector( selector ) );
 			}
 		} );
-	}
+		observer.observe( container, {
+			childList: true,
+			subtree: true,
+		} );
+
+		// Stop trying after 3 seconds.
+		timer = setTimeout( () => {
+			observer.disconnect();
+			resolve( null );
+		}, 3000 );
```

Comment:
Not sure if we want to make this configurable. I guess it will work as a default value for now.